### DISCUSSION TITLE
What is the role of async in `await await act(async () => { render(<App />) });`

### DISCUSSION THREAD

**Discussion by bybrunobarros** [react-hook-form/react-hook-form#8712] [Q&A]
Hey everyone 👋,

While I was working on some issues around out tests I stumbled on this documentation ([Testing Form](https://react-hook-form.com/advanced-usage#TestingForm) → Resolving act warning during test).
I'm trying to understand if I can get rid of some issues thanks to the documentation. And there is something I can't figure out. What is the purpose of async in this code ?
```javascript
await act(async () => {
  render(<App />)
});
```
I just can't figure it out. Could you help me to understand ? Like is it an issue to not using it and why ?

Thanks.

**Comment by Moshyfawn** [comment]
Check out #7125 discussion and see if it helps

**Reply by bybrunobarros** [reply]
Thank you for pointing that out. It doesn't really answer my question, or I'm maybe missing something 🤔. I understand well that `act` returns `Promise<void>`.
What I don't understand is: why the function used by `act` is defined as an `async` ?
I tried with and without `async`, I didn't find any difference. 
I even tried to reproduce something similar, and didn't find any reason to use `async` there.

```javascript
const it = (str, test) => {
  return test(str)
}
const act = (fn) => fn();

const delay = (time) => new Promise((resolve)=> setTimeout(resolve, time));

const delayedFn = async (time) => {
  console.log("before delay");
  await delay(time);
  console.log("after delay");
};

it("some test case", async (title) => {
   await act(/* async */ () => {
     delayedFn(5000);
     console.log(title);
  });
});
```

It's probably not that important, but I can't get it out of my mind 😅
